server/src/main/java/org/apache/druid/server/coordinator/duty/RunRules.java

```diff
@@ -46,38 +45,16 @@ public class RunRules implements CoordinatorDuty
   private static final EmittingLogger log = new EmittingLogger(RunRules.class);
   private static final int MAX_MISSING_RULES = 10;
 
-  private final ReplicationThrottler replicatorThrottler;
+  private final SegmentStateManager stateManager;
 
-  private final DruidCoordinator coordinator;
-
-  public RunRules(DruidCoordinator coordinator)
```

Comment:
Change summary:
Move replication throttling logic to `SegmentStateManager`.
Move check on `maxNonPrimaryReplicantsToLoad` to `ReplicationThrottler` (requires revisit).